First attempt for making rasa run listen on a specific ip

bastbnl opened the pull request:
**Proposed changes**:
- Restrict the network interfaces used by rasa to one specific address instead of `0.0.0.0`

I updated one single document and I'm happy to update other files as needed.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RasaHQ/rasa?pullRequest=8470) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RasaHQ/rasa?pullRequest=8470) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RasaHQ/rasa?pullRequest=8470) it.</sub>

sara-tagger commented:
Thanks for submitting a pull request 🚀 @degiz will take a look at it as soon as possible ✨

joejuzl reviewed (changes requested):
Thanks for tackling this @bastbnl!

I think we should handle the host in the same way we currently handle `port`.
See `add_port_argument` in `rasa/cli/arguments/run.py`.

bastbnl commented:
> 
> 
> Thanks for tackling this @bastbnl!
> 
> I think we should handle the host in the same way we currently handle `port`.
> See `add_port_argument` in `rasa/cli/arguments/run.py`.

Thanks for getting back to me, @joejuzl. I think I added the changes you requested and I must say that it beats the `SANIC_HOST` approach. I added some doc updates as well. 

The `SANIC_HOST` approach was merged into the `rasa-sdk` package, so merging this PR does create an inconsistency

joejuzl commented on CHANGELOG.mdx:
```diff
@@ -16,6 +16,122 @@ https://github.com/RasaHQ/rasa/tree/main/changelog/ . -->
 
 <!-- TOWNCRIER -->
 
+## [2.8.0] - 2021-07-12
```
Looks like you need to make sure your base branch is up-to-date. These changes shouldn't be here.

joejuzl commented on docs/docs/messaging-and-voice-channels.mdx:
```diff
@@ -59,3 +59,13 @@ With the free-tier of ngrok, you can run into limits on how many connections you
 
 :::
+
+Alternatively you can make your assistent listen on a specific address using the `-i` command line 
```
Can we also add something to `docs/docs/command-line-interface.mdx`?

bastbnl replied:
Did that!

joejuzl commented on rasa/cli/arguments/run.py:
```diff
@@ -32,6 +32,19 @@ def add_port_argument(
+    """Binds the RASA process to a network interface."""
+    parser.add_argument(
+        "-i",
+        "--interface",
```
I feel like `--host` might be more idiomatic.

bastbnl replied:
Tried using that when I started working on it, but found an issue with the short argument: `-h`, which is generally used for help. Then I thought about adding `-i` and `--host`, but that looked even more confusing so eventually settled on `-i`. Happy to change it though.

joejuzl replied:
Good point - let's keep it as `-i`

joejuzl commented on rasa/core/constants.py:
```diff
@@ -1,8 +1,10 @@
 DEFAULT_SERVER_PORT = 5005
 
-DEFAULT_SERVER_FORMAT = "{}://localhost:{}"
+DEFAULT_SERVER_INTERFACE = "0.0.0.0"
```
hmm so this was `localhost` before, but we actually set the listen host to `0.0.0.0` - which does seem like it was wrong. However I'm slightly worried it may have consequence as we use this for things other than just logging. I _think_ it's most likely fine though...

bastbnl replied:
Well, it was `localhost`-in-logging-line-only. Code showed it was actually listening on `0.0.0.0`

bastbnl replied:
> 
> 
> Thanks for changing the approach so quickly. 🚀
> 
> Good point about the inconsistency with the SDK, I hadn't realised that - but I think it's probably fine, we can add a similar flag to the action server at some point.
> 
> It seems like you may have to rebase your branch, as some unrelated changes leaking in.

I've been trying to rebase this branch with main all evening and I've not been able to get it done, mainly due to my understanding of git and the vscode tooling I'm using. Any way to get this done without rebasing, @joejuzl? 

> 
> Also could you update the tests at `tests/cli/test_rasa_run.py`?

I'm working on a PR for rasa-sdk which supports both an option and the environment variable, since I contributed that.

joejuzl replied:
> Well, it was localhost-in-logging-line-only. Code showed it was actually listening on 0.0.0.0

`DEFAULT_SERVER_FORMAT` is used in `DEFAULT_SERVER_URL` which is used in a couple other places. But I don't think it will cause an issue.

> I've been trying to rebase this branch with main all evening and I've not been able to get it done, mainly due to my understanding of git and the vscode tooling I'm using. Any way to get this done without rebasing, @joejuzl?

I think the problem is the commit `Merge remote-tracking branch 'upstream/2.8.x' into bastbnl-issue-8469`. To fix this you either have to do an interactive rebase onto `main` and skip that commit. Or you can merge in `origin/main` and then `git reset --soft origin/main` (replacing `origin` with whatever the remote called for you) - this will leave all your changes staged but uncommited, on top of master, but you will loose your commit history.

joejuzl reviewed (changes requested):
Thanks for changing the approach so quickly.  🚀 

Good point about the inconsistency with the SDK, I hadn't realised that - but I think it's probably fine, we can add a similar flag to the action server at some point.

It seems like you may have to rebase your branch, as some unrelated changes leaking in.

Also could you update the tests at `tests/cli/test_rasa_run.py`?

bastbnl commented:
Whoops... I think I basically messed up. Maybe it's a better idea to create a new PR?

joejuzl commented:
> Whoops... I think I basically messed up. Maybe it's a better idea to create a new PR?

@bastbnl if you find that easier, go for it. You can request a review from me directly on it.

bastbnl commented:
> 
> 
> > Whoops... I think I basically messed up. Maybe it's a better idea to create a new PR?
> 
> @bastbnl if you find that easier, go for it. You can request a review from me directly on it.

Did that in PR #9242, @joejuzl. Thanks for your guidance here. I didn't close the PR as I don't know what the consequences of closing it would be on your end.

joejuzl commented:
We can close this one and I will look at the new PR.
